Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -540,39 +580,58 @@ void DrawFrame() {
-                    renderPassDescriptor.colorAttachments[0].clearColor = MTLClearColorMake(0.0,0.0,0.0,1.0);
-                    
-                    // Create a render command encoder.
-                    id<MTLRenderCommandEncoder> renderEncoder = [commandBuffer renderCommandEncoderWithDescriptor:renderPassDescriptor];
-                    renderEncoder.label = @"XRDisplayEncoder";
+                        // Set the region of the drawable to draw into.
+                        [renderEncoder setViewport:(MTLViewport){0.0, 0.0, static_cast<double>(viewportSize.x), static_cast<double>(viewportSize.y), 0.0, 1.0 }];
+                        
+                        [renderEncoder setRenderPipelineState:pipelineState];
 
-                    // Set the region of the drawable to draw into.
-                    [renderEncoder setViewport:(MTLViewport){0.0, 0.0, static_cast<double>(viewportSize.x), static_cast<double>(viewportSize.y), 0.0, 1.0 }];
-                    
-                    [renderEncoder setRenderPipelineState:pipelineState];
+                        // Pass in the parameter data.
+                        [renderEncoder setVertexBytes:vertices length:sizeof(vertices) atIndex:0];
 
-                    // Pass in the parameter data.
-                    [renderEncoder setVertexBytes:vertices length:sizeof(vertices) atIndex:0];
+                        [renderEncoder setFragmentTexture:id<MTLTexture>(ActiveFrameViews[0].ColorTexturePointer) atIndex:0];
+                        [renderEncoder setFragmentTexture:cameraTextureY atIndex:1];
+                        [renderEncoder setFragmentTexture:cameraTextureCbCr atIndex:2];
 
-                    [renderEncoder setFragmentTexture:id<MTLTexture>(ActiveFrameViews[0].ColorTexturePointer) atIndex:0];
-                    [renderEncoder setFragmentTexture:sessionDelegate.cameraTextureY atIndex:1];
-                    [renderEncoder setFragmentTexture:sessionDelegate.cameraTextureCbCr atIndex:2];
+                        // Draw the triangles.
+                        [renderEncoder drawPrimitives:MTLPrimitiveTypeTriangleStrip vertexStart:0 vertexCount:4];
 
-                    // Draw the triangles.
-                    [renderEncoder drawPrimitives:MTLPrimitiveTypeTriangleStrip vertexStart:0 vertexCount:4];
+                        [renderEncoder endEncoding];
 
-                    [renderEncoder endEncoding];
+                        // Schedule a present once the framebuffer is complete using the current drawable.
+                        [commandBuffer presentDrawable:drawable];
+                    }
 
-                    // Schedule a present once the framebuffer is complete using the current drawable.
-                    [commandBuffer presentDrawable:drawable];
+                    // Finalize rendering here & push the command buffer to the GPU.
+                    [commandBuffer commit];
+                    [commandBuffer waitUntilCompleted];
```

Comment:
Could this cause a perf issue?

Reply:
Replaced with a completion handler to free the textures after the command buffer is exhausted.